### PR TITLE
[WIP] `@test_reference`: add the `return_bool` keyword argument to get the pass/fail status as a `Bool`

### PR DESCRIPTION
**Pull request by DilumAluthge**

### CODE REVIEW

File: src/test_reference.jl [hunk 1]
```diff
@@ -117,6 +117,13 @@ function test_reference(
     rendermode=nothing;
     kw...) where {F <: DataFormat, T}
 
+    if haskey(kw, :return_bool)
+        return_bool = kw[:return_bool]::Bool
+    else
+        return_bool = false
+    end
+    kw = filter(p -> p[1] != :return_bool, kw)
```
Comment (IanButterworth) [inline src/test_reference.jl:125]: Just a suggestion for simpler code
```suggestion
    return_bool  = get(kw, :return_bool, false)::Bool
    delete!(kw, :return_bool)
```

**Comment by IanButterworth** [comment]
This might be better as a `@check_reference` macro, or something like that? 
`@test...` is kind of tied into the Test architecture?

Not a strong opinion

File: src/test_reference.jl (hunk 1 above)
Comment (DilumAluthge) [reply src/test_reference.jl:125]: The `return_bool = get(kw, :return_bool, false)::Bool` part is good.

The `delete!` throws a `MethodError`.

**Comment by johnnychen94** [comment]
This is a good change but might need some refactoring work. We've discussed changing the interface to `match_reference(x, ref; kw...)::Bool` would be a satisfying solution: that people can use normal Tests utilities to do the common work, e.g.,

```julia
@test match_reference(line, "this is the reference line")
@test_broken match_reference(line, "this is another line")
```

**Comment by DilumAluthge** [comment]
Sure, I can refactor the code out into a `match_reference` function, and then `test_reference` would call `match_reference` under the hood.

**Comment by DilumAluthge** [comment]
#112

**Pull request closed without merging**